platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js

```diff
@@ -26,19 +27,55 @@ class ViewerRetrieveStudyData extends Component {
 
   async loadStudies() {
     try {
-      const { server, studyInstanceUids, seriesInstanceUids } = this.props;
+      const { server, studyInstanceUids, seriesInstanceUIDs } = this.props;
+      const filters = {};
+
+      // Use the first, discard others
+      const seriesInstanceUID = seriesInstanceUIDs && seriesInstanceUIDs[0];
+
+      if (seriesInstanceUID) {
+        filters.seriesInstanceUID = seriesInstanceUID;
+      }
+
       const studies = await retrieveStudiesMetadata(
         server,
         studyInstanceUids,
-        seriesInstanceUids
+        filters
       );
+      this.validateFilters(studies, filters);
       this.setStudies(studies);
     } catch (e) {
       this.setState({ error: true });
       log.error(e);
     }
   }
 
+  /**
+   * Validate filters and promp user a message in case filter is unsuccessfully applied.
+   * In case of success, studies array contains, as the first element, the queried content (from filter)
+   * @param {Array} studies array of studies to be evaluated
+   * @param {Object} filters filters to test against
+   */
+  validateFilters(studies = [], filters = {}) {
+    const { seriesInstanceUID } = filters;
+
+    const { snackbarContext } = this.props;
+    // skip in case no filter or no toast manager
+    if (!seriesInstanceUID || !snackbarContext) {
+      return;
+    }
+
+    const firstStudy = studies[0] || {};
+    const { seriesList = [] } = firstStudy;
+    const firstSeries = seriesList[0];
+
+    if (!firstSeries || firstSeries.seriesInstanceUid !== seriesInstanceUID) {
```

Comment:
Our "filter validation" is really a test to see that _any_ results were returned after we've applied our filters to our query, right? It might be easier to just check that we returned a study with one or more series?

Reply:
not actually because as per requirement, it will always return series (filtered or not)..

Reply:
Ah. Okay. So this is further down the pipe. We've hit our fallback (study retrieve), so we're more or less guaranteed to have data. We have to check the prop to see if filtering was applied.

What if the first series in the list just happens to match our filter?